internal/integration/provision/upgrade.go

```diff
@@ -133,29 +133,33 @@ func upgradeCurrentToCurrent() upgradeSpec {
 	}
 }
 
-// upgradeStableToCurrentPreserve upgrades from the stable Talos release to the current version for single-node cluster with preserve.
-func upgradeStableToCurrentPreserve() upgradeSpec {
+// upgradeCurrentToCurrentBios upgrades the current version to itself without UEFI.
+func upgradeCurrentToCurrentBios() upgradeSpec {
+	installerImage := fmt.Sprintf(
+		"%s/%s:%s",
+		DefaultSettings.TargetInstallImageRegistry,
+		images.DefaultInstallerImageName,
+		DefaultSettings.CurrentVersion,
+	)
+
 	return upgradeSpec{
-		ShortName: fmt.Sprintf("prsrv-%s-%s", stableRelease, DefaultSettings.CurrentVersion),
+		ShortName: fmt.Sprintf("%s-same-ver-bios", DefaultSettings.CurrentVersion),
 
-		SourceKernelPath:     helpers.ArtifactPath(filepath.Join(trimVersion(stableRelease), constants.KernelAsset)),
-		SourceInitramfsPath:  helpers.ArtifactPath(filepath.Join(trimVersion(stableRelease), constants.InitramfsAsset)),
-		SourceInstallerImage: fmt.Sprintf("%s:%s", "ghcr.io/siderolabs/installer", stableRelease),
-		SourceVersion:        stableRelease,
-		SourceK8sVersion:     stableK8sVersion,
+		SourceKernelPath:     helpers.ArtifactPath(constants.KernelAssetWithArch),
```

Comment:
do we need to start with a disk image, or this would have already reproduced the bug you fixed?

Reply:
It might not, I'll confirm once I'm back

Reply:
it does work, so all good

Reply:
actually scratch that, I did a wrong test, will do a different PR to fix